src/main/java/uk/gov/hmcts/reform/managecase/repository/DefaultDataStoreRepository.java

```diff
@@ -208,9 +195,22 @@ private void setChangeOrganisationRequestApprovalStatus(ChangeOrganisationReques
     }
 
     @Override
-    public CaseDetails findCaseByCaseIdExternalApi(String caseId) {
+    public CaseDetails findCaseByCaseIdUsingExternalApi(String caseId) {
+        return findCaseByCaseIdExternalApi(caseId, false);
+    }
+
+    @Override
+    public CaseDetails findCaseByCaseIdAsSystemUserUsingExternalApi(String caseId) {
+        return findCaseByCaseIdExternalApi(caseId, true);
+    }
+
+    private CaseDetails findCaseByCaseIdExternalApi(String caseId, boolean asSystemUser) {
         try {
-            return dataStoreApi.getCaseDetailsByCaseIdViaExternalApi(caseId);
+            if (asSystemUser) {
+                return dataStoreApi.getCaseDetailsByCaseIdViaExternalApi(securityUtils.getCaaSystemUserToken(), caseId);
```

Comment:
better to do this in findCaseByCaseIdAsSystemUserUsingExternalApi, instead of passing boolean values